src/durable-cache/src/lib.rs

```diff
@@ -340,6 +338,46 @@ impl<C: DurableCacheCodec> DurableCache<C> {
 
         Ok(new_upper)
     }
+
+    /// Forcibly compacts the shard backing this cache in a background task. See
+    /// [`mz_persist_client::cli::admin::dangerous_force_compaction_and_break_pushdown`].
+    pub fn dangerous_compact_shard(
+        &self,
+        persist: PersistClient,
+        fuel: impl Fn() -> usize + Send + 'static,
+        wait: impl Fn() -> Duration + Send + 'static,
+    ) -> JoinHandle<()> {
+        let shard_id = self.write.shard_id();
+        let (key_schema, val_schema) = C::schemas();
+        let purpose = format!("{}_compaction", self.purpose);
+        spawn(
+            || format!("durable-cache-compaction-{}", self.purpose),
+            async move {
+                let write: WriteHandle<C::KeyCodec, C::ValCodec, u64, i64> = persist
+                    .open_writer(
```

Comment:
Once we do the other comment to pull this out of the background task, then we can use `self.write` instead of opening a new WriteHandle I think?